features/nationaliteit.feature

```diff
@@ -1,59 +1,301 @@
+      | fields              | burgerservicenummer,nationaliteiten |
+      Dan heeft de persoon met burgerservicenummer '555550002' een nationaliteit met alleen de volgende gegevens
+      | naam                  | waarde   |
+      | nationaliteit         | 0263     |
+      | redenOpname           | 301      |
+      | datumIngangGeldigheid | 20100801 |
+      En heeft de persoon met burgerservicenummer '555550002' GEEN nationaliteit met de volgende gegevens
+      | naam                  | waarde   |
+      | datumIngangGeldigheid | 20100801 |
+      En heeft de persoon met burgerservicenummer '555550002' GEEN nationaliteit met de volgende gegevens
+      | naam          | waarde |
+      | nationaliteit | 0100   |
+      | redenOpname   | 301    |
+
+    @gba
+    Scenario: geactualiseerde ingangsdatum geldigheid
+      Gegeven de persoon met burgerservicenummer '555550003' heeft de volgende gegevens
+      | Stapel | Categorie | nationaliteit (05.10) | reden opnemen (63.10) | reden beëindigen (64.10) | bijzonder Nederlanderschap (65.10) | onjuist (84.10) | datum ingang geldigheid (85.10) |
+      | 1      | 4         | 0100                  | 311                   |                          |                                    |                 | 20200727                        |
+      | 1      | 54        | 0100                  | 311                   |                          |                                    | O               | 20200713                        |
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                              |
+      | type                | RaadpleegMetBurgerservicenummer     |
+      | burgerservicenummer | 555550003                           |
+      | fields              | burgerservicenummer,nationaliteiten |
+      Dan heeft de persoon met burgerservicenummer '555550003' een nationaliteit met alleen de volgende gegevens
+      | naam                  | waarde   |
+      | nationaliteit         | 0100     |
+      | redenOpname           | 311      |
+      | datumIngangGeldigheid | 20200727 |
+
+  @proxy
+  Rule: de nationaliteit wordt gevuld op basis van het voorkomen van nationaliteit of aanduidingBijzonderNederlanderschap
+    - wanneer nationaliteit voorkomt met een waarde ongelijk aan "0000" (onbekend), dan wordt type opgenomen met de waarde "Nationaliteit"
+    - wanneer nationaliteit voorkomt met een waarde gelijk aan "0000" (onbekend), dan wordt type opgenomen met de waarde "NationaliteitOnbekend"
+    - wanneer aanduidingBijzonderNederlanderschap voorkomt met een waarde gelijk aan "B", dan wordt type opgenomen met de waarde "BehandeldAlsNederlander"
+    - wanneer aanduidingBijzonderNederlanderschap voorkomt met een waarde gelijk aan "V", dan wordt type opgenomen met de waarde "VastgesteldNietNederlander"
+
+    @proxy
+    Scenario: de persoon heeft de Nederlandse nationaliteit
```

Comment:
Even kijken of ik het snap. Volgens de rule zou iemand een nationaliteit (type=nationaliteit) hebben als nationaliteit.code ongelijk is aan 0000 en (waarschijnlijk) aanduidingBijzonderNederlanderschap heeft geen waarde.
Is het dan in deze scenario niet voldoende om in de gegeven stap alleen nationaliteit.code = 0001 op te nemen? redenOpname en datumIngangGeldigheid is dan toch niet relevant?

Reply:
Door meer op te nemen in de specificatie dan nodig is om de nationaliteit type te bepalen, bezorgt je jezelf ook meer werk bij fouten. In alle scenario's moet in de gegeven stap 'nationaliteit' 'nationaliteit.code' zijn en 'redenOpname' 'redenOpname.code'. Hetzelfde geldt ook in de Dan stap en moet datumIngangGeldigheid naar de polymorfe type.

Waarom is datumIngangGeldigheid niet gespecificeerd in datum.feature? Ik ben op zoek naar wanneer een gegeven wordt gespecificeerd bij zijn gegevensgroep/categorie zoals hier of generiek in bijv. de datum/in_onderzoek feature

Reply:
Persoonlijk zou ik ervoor kiezen om het gegevensgroep/categorie te specificeren. Nu is de datum feature pas done als alle groepen (geboorte, overlijden, verblijfplaats, etc) zijn geïmplementeerd

Reply:
ik verwijder overbodige gegevens uit de gegeven en dan stappen

Reply:
> Persoonlijk zou ik ervoor kiezen om het gegevensgroep/categorie te specificeren. Nu is de datum feature pas done als alle groepen (geboorte, overlijden, verblijfplaats, etc) zijn geïmplementeerd

Dit was alleen opgenomen ter illustratie. Mijn idee was dat waneer een materiedeskundige dit zou lezen deze het gek zou vinden dat er alleen een nationaliteit is en geen reden opname of datum ingang. Maar ik heb beide nu verwijderd.